x-pack/plugins/security_solution/public/common/store/sourcerer/readme.md

```diff
@@ -0,0 +1,80 @@
+# Sourcerer Redux
+
+Sourcerer model for redux
+```typescript
+interface SourcererModel {
+  /** default security-solution data view */
+  defaultDataView: SourcererDataView & { id: string; error?: unknown };
+  /** all Kibana data views, including security-solution */
+  kibanaDataViews: SourcererDataView[];
+  /** security solution signals index name */
+  signalIndexName: string | null;
+  /** sourcerer scope data by id */
+  sourcererScopes: SourcererScopeById;
+}
+```
+
+The SourcererScopeName uniquely identifies a Sourcerer Scope. There are 3 in our app:
+```typescript
+enum SourcererScopeName {
+  default = 'default',
+  detections = 'detections',
+  timeline = 'timeline',
+}
+```
+
+Data related to each sourcerer scope
+```typescript
+interface SourcererScope {
+  /** Uniquely identifies a Sourcerer Scope */
+  id: SourcererScopeName;
+  /** is an update being made to the sourcerer data view */
+  loading: boolean;
+  /** selected data view id, null if it is legacy index patterns*/
+  selectedDataViewId: string | null;
+  /** selected patterns within the data view */
+  selectedPatterns: string[];
+  /** if has length,
+   * id === SourcererScopeName.timeline
+   * selectedDataViewId === null OR defaultDataView.id
+   * saved timeline has pattern that is not in the default */
+  missingPatterns: string[];
+}
+
+type SourcererScopeById = Record<SourcererScopeName, SourcererScope>;
+```
+
+```typescript
+interface KibanaDataView {
+  /** Uniquely identifies a Kibana Data View */
+  id: string;
+  /**  list of active patterns that return data  */
+  patternList: string[];
+  /**
+   * title of Kibana Data View
+   * title also serves as "all pattern list", including inactive
+   * comma separated string
+   */
+  title: string;
+}
+```
+
+KibanaDataView + timelines/index_fields enhanced field data
+```typescript
+interface SourcererDataView extends KibanaDataView {
+  id: string;
+  /** we need this for @timestamp data */
```

Comment:
This comment doesn't seem to say much about `browserFields` and `docValueFields` properties.